src/desdemona/query.clj

```diff
@@ -53,3 +61,24 @@
    (run-logic-query (dsl->logic dsl-query) events))
   ([n-answers dsl-query events]
    (run-logic-query n-answers (dsl->logic dsl-query) events)))
+
+(def ^:private infix-parser
+  (insta/parser (resource "infix-query-grammar.ebnf")))
+
+(defn ^:private parsed-infix->dsl
+  [parsed]
+  (m/match parsed
+    [:expr terms]
+    (parsed-infix->dsl terms)
+
+    [:eq
+     [:fn-call
+      [:identifier "ip"]
+      [:identifier arg]]
+     [:ipv4-address & addr-parts]]
+    (let [arg (symbol arg)
+          addr (s/join "." addr-parts)]
+      `(~'= (:ip ~arg) ~addr))))
```

Comment:
Yes. I am not sure which parts of the behavior should live in the grammar and which in this compily bit. I'm inclined to want it to be as early as possible; but writing compiler semantics here is _way_ easier than doing them in the grammar, and the grammar already means that we have a real parser and not a weird machine.